Makefile

```
@@ -1,321 +1,259 @@
-
+	./hack/image/build-scorecard-test-kuttl-image.sh $(SCORECARD_TEST_KUTTL_CI_IMAGE):dev
 image-push-scorecard-test-kuttl:
-	./hack/image/push-image-tags.sh $(SCORECARD_TEST_KUTTL_BASE_IMAGE):dev $(SCORECARD_TEST_KUTTL_IMAGE)-$(shell go env GOARCH)
-
+	./hack/image/push-image-tags.sh $(SCORECARD_TEST_KUTTL_CI_IMAGE):dev $(SCORECARD_TEST_KUTTL_IMAGE)-$(shell go env GOARCH)
 image-push-scorecard-test-kuttl-multiarch:
 	./hack/image/push-manifest-list.sh $(SCORECARD_TEST_KUTTL_IMAGE) ${SCORECARD_TEST_KUTTL_ARCHES}
 
-##############################
-# Tests                      #
-##############################
-
-##@ Tests
+.PHONY: image-build-custom-scorecard-tests
+CUSTOM_SCORECARD_TESTS_CI_IMAGE = quay.io/operator-framework/custom-scorecard-tests
+CUSTOM_SCORECARD_TESTS_IMAGE ?= $(CUSTOM_SCORECARD_TESTS_CI_IMAGE)
+image-build-custom-scorecard-tests:
+	./hack/image/build-custom-scorecard-tests-image.sh $(CUSTOM_SCORECARD_TESTS_CI_IMAGE):dev
 
-# Static tests.
-.PHONY: test test-sanity test-unit
+##@ Test
 
-test: test-unit ## Run the tests
+.PHONY: test-all
+test-all: test-static test-e2e ## Run all tests
 
-test-sanity: tidy build/operator-sdk lint
-	./hack/tests/sanity-check.sh
+.PHONY: test-static
+test-static: test-sanity test-unit test-links ## Run all non-cluster-based tests
 
-test-unit: ## Run the unit tests
-	$(Q)go test -coverprofile=coverage.out -covermode=count -count=1 -short $(TEST_PKGS)
+.PHONY: test-sanity
+test-sanity: generate fix ## Test repo formatting, linting, etc.
+	git diff --exit-code # fast-fail if generate or fix produced changes
+	./hack/check-license.sh
+	./hack/check-error-log-msg-format.sh
+	go run ./hack/generate/changelog/gen-changelog.go -validate-only
+	go vet ./...
+	./tools/scripts/fetch golangci-lint 1.31.0 && ./tools/bin/golangci-lint run
+	git diff --exit-code # diff again to ensure other checks don't change repo
 
-test-links:
+.PHONY: test-links
+test-links: ## Test doc links
 	./hack/check-links.sh
 
-# CI tests.
-.PHONY: test-ci
-
-test-ci: test-sanity test-unit install test-subcommand test-e2e ## Run the CI test suite
-
-# Subcommand tests.
-.PHONY: test-subcommand test-subcommand-olm-install
-
-test-subcommand: test-subcommand-olm-install
-
-test-subcommand-olm-install:
+.PHONY: test-unit
+TEST_PKGS = $(shell go list ./... | grep -v -E 'github.com/operator-framework/operator-sdk/test/')
+test-unit: ## Run unit tests
+	go test -coverprofile=coverage.out -covermode=count -short $(TEST_PKGS)
+
+e2e_tests := test-e2e-go test-e2e-ansible test-e2e-ansible-molecule test-e2e-helm test-e2e-integration
+e2e_targets := test-e2e $(e2e_tests)
+.PHONY: $(e2e_targets)
+
+.PHONY: test-e2e-setup
+export KIND_CLUSTER := operator-sdk-e2e
+export KUBECONFIG := $(HOME)/.kube/kind-$(KIND_CLUSTER).config
+export KUBEBUILDER_ASSETS := $(PWD)/tools/bin
+test-e2e-setup: build
+	./tools/scripts/fetch kind 0.9.0
+	./tools/scripts/fetch kubectl ${K8S_VERSION}
+	./tools/scripts/fetch envtest 0.6.3
```

Something I just remembered: the envtest tarball contains `kubectl`, so if we want the specified `kubectl` version then switch these lines:
```suggestion
	./tools/scripts/fetch envtest 0.6.3
	./tools/scripts/fetch kubectl $(K8S_VERSION)
```

---

Oh man good catch. I'll flip these and add a comment about this.